### PR TITLE
GDB-4864 IRIs in preview to be links to resource

### PR DESCRIPTION
**Pull request by yordanalexandrov**
If the cell contains an IRI, it is made a link to the resource in the DB

### CODE REVIEW

File: src/app/main/mapper/cell/cell.component.ts [hunk 1]
```diff
@@ -269,4 +270,25 @@ export class CellComponent extends OnDestroyMixin implements OnInit {
...
     return !this.triple.isEmpty() && this.isEmpty();
   }
+
+  getResourceUri(previewItem: string): any {
+    let uri;
+    if (previewItem.match(/^<|>$/g)) {
+      previewItem = previewItem.replace(/^<|>$/g, '');
+    }
+    if (this.getTransformation() && this.getTransformation().getLanguage() === RAW_CONSTANT) {
+      if (previewItem.indexOf(this.baseIRI) > -1) {
```
Comment (svilenvelikov) [inline src/app/main/mapper/cell/cell.component.ts:280]: `indexOf` doesn't mean exact match, is that OK?

File: src/app/main/mapper/cell/preview-cell/preview-cell.component.html [hunk 2]
```diff
@@ -24,8 +24,14 @@
           </div>
 
           <div  *ngIf="isFirstChild && !isTypeProperty; else emptyBlock">
-            <div fxFill [ngClass]="['ti-preview', (isEmptyPreview() ? 'empty-preview' : '')]" *ngFor="let previewItem of getPreview();">
-              {{previewItem}}
+            <div fxFill [ngClass]="['ti-preview', (isEmptyPreview() ? 'empty-preview' : '')]"
+                 *ngFor="let previewItem of getPreview();">
+              <ng-container *ngIf="this.getType() === IRI; else notIRI">
```
Comment (svilenvelikov) [inline src/app/main/mapper/cell/preview-cell/preview-cell.component.html:29]: I hate how much duplication there is in this application :(

File: src/app/main/mapper/iteration/iteration.component.html [hunk 3]
```diff
@@ -1,70 +1,73 @@
 <div appCypressData="mapping-wrapper">
-  <div *ngFor="let mapping of triples; let index = index" fxLayout="row" fxLayoutAlign="start stretch"
+  <div *ngFor="let triple of triples; let index = index" fxLayout="row" fxLayoutAlign="start stretch"
```
Comment (svilenvelikov) [inline src/app/main/mapper/iteration/iteration.component.html:2]: :+1: 
I like the boy scout appproach you have :)

**Review by svilenvelikov — APPROVED** [review]

**Review by desislava-hristova-ontotext — APPROVED** [review]

**Review by teodossidossev — APPROVED** [review]

File: src/app/main/mapper/cell/cell.component.ts (hunk 1 above)
Comment (yordanalexandrov) [reply src/app/main/mapper/cell/cell.component.ts]: Removed the check as it turned out that it is not needed when using raw IRI

File: src/app/main/mapper/cell/configuration-and-preview-cell/configuration-and-preview-cell.component.html [hunk 4]
```diff
@@ -166,17 +166,26 @@
...
-      <div class="ti-footer" *ngIf="isFirstChild && nestLevel === 0">
-        <div [ngClass]="['ti-preview', (isEmptyPreview() ? 'empty-preview' : '')]">
-          <div *ngFor="let previewItem of getPreview();">
-            {{previewItem}}
-          </div>
-          <div *ngIf="isFirstChild && isTypeProperty">
-            <div>a</div>
+      <div *ngIf="isFirstChild && !isTypeProperty; else emptyBlock">
+        <div class="ti-footer" *ngIf="isFirstChild && nestLevel === 0">
+          <div *ngIf="isFirstChild && !isTypeProperty; else emptyBlock">
```
Comment (svilenvelikov) [inline src/app/main/mapper/cell/configuration-and-preview-cell/configuration-and-preview-cell.component.html:171]: @teodossidossev @desislava-hristova-ontotext I hope this nesting here is OK

**Review by svilenvelikov — APPROVED** [review]

File: src/app/main/mapper/cell/preview-cell/preview-cell.component.html (hunk 2 above)
Comment (yordanalexandrov) [reply src/app/main/mapper/cell/preview-cell/preview-cell.component.html:29]: I will play around with templates to see if I can get a nice way to reduce view duplication.

File: src/app/main/mapper/cell/configuration-and-preview-cell/configuration-and-preview-cell.component.html (hunk 4 above)
Comment (yordanalexandrov) [reply src/app/main/mapper/cell/configuration-and-preview-cell/configuration-and-preview-cell.component.html]: I refactored the view code. No more unnecessary nesting.

**Pull request merged**